src/sentry/models/file.py

```diff
@@ -431,6 +431,12 @@ def assemble_from_file_blob_ids(self, file_blob_ids, checksum, commit=True):
         tf.seek(0)
         return tf
 
+    def delete(self, *args, **kwargs):
+        blob_ids = [blob.id for blob in self.blobs.all()]
+        super(File, self).delete(*args, **kwargs)
+
+        delete_unreferenced_blobs.delay(blob_ids)
```

Comment:
It might be worth doing this with `transaction.on_commit`. I don't know whether delete here is running in a transaction, but if it is then it's possible that `delete_unreferenced_blobs` could run before this transaction commits, and fail to remove the rows.  I've only started using this recently, but there's an example in the codebase.